vg/vg_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vg_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/plotter"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+	"github.com/gonum/plot/vg/vgeps"
+	"github.com/gonum/plot/vg/vgimg"
+	"github.com/gonum/plot/vg/vgpdf"
+	"github.com/gonum/plot/vg/vgsvg"
+)
+
+// TestLineWidth tests output against test images generated by
+// gen_width.go - these images have been visually inspected and
+// validated.
+func TestLineWidth(t *testing.T) {
+	formats := []string{
+		// TODO: Add logic to cope with run to run eps differences.
+		"pdf",
+		"svg",
+		"png",
+		"tiff",
+		"jpg",
+	}
+
+	const (
+		width  = 100
+		height = 100
+	)
+
+	for _, w := range []vg.Length{-1, 0, 1} {
```

Comment:
How would you feel about parsing the width and format out of the testdata file name? Then if you add new formats you don't have to modify both the gen script and this file.